File: 1318-minimum-flips-to-make-a-or-b-equal-to-c/1318-minimum-flips-to-make-a-or-b-equal-to-c.cpp
class Solution {
public:
    int minFlips(int a, int b, int c) {
        int ans=0;
        
        
        for(int i=0;i<32;i++){
            bool ai=false;
            bool bi=false;
            bool ci=false;
            if((a & (1<<i))>0) ai=true; //setting the ith bits
            if((b & (1<<i))>0) bi=true;
            if((c & (1<<i))>0) ci=true;
            
            if(ci){
                if(!ai and !bi) ans++; //if both bits are unset
            }
            else{
                if(ai and bi) ans+=2; //if both are set
                else if(ai || bi) ans+=1; //if either is set
            }
        }
        return ans;
    }
};

File: 1318-minimum-flips-to-make-a-or-b-equal-to-c/1318-minimum-flips-to-make-a-or-b-equal-to-c.py
class Solution:
    def minFlips(self, a: int, b: int, c: int) -> int:
        count = 0
        while(a or b or c):
            bit_a, bit_b, bit_c = a&1, b&1, c&1
            if(bit_a | bit_b != bit_c):
                if( bit_a == 1 and bit_b == 1):
                    count += 2
                else:
                    count += 1
            a = a>>1
            b = b>>1
            c = c>>1
        return count